DirectX11/globals.h

```diff
@@ -11,13 +11,21 @@
 
 #include "DLLMainHook.h"
 #include "DirectXMath.h"
-#include "util.h"
 #include "DecompileHLSL.h"
 
 #include "ResourceHash.h"
 #include "CommandList.h"
 #include "profiling.h"
 #include "lock.h"
+#include <enum.h>
+
+#include "../HackingCommon/globals_common.h"
+#define ResourceSnapshot Snapshot<ID3D11Resource>
+#define ShaderInfoData InfoData<ID3D11Resource>
+#define TargetResourceSelection TargetResourceSelection<ID3D11Resource*, ID3D11Resource, ResourceHashInfo>
+#define ShaderResourceSelection_(type) ShaderResourceSelection<UINT64, ID3D11Resource, type>
```

Comment:
I'm not certain I like these typedefs masquerading as defines. Looks dangerous, please justify.